generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -22,13 +22,42 @@ def __init__(self, grpc_options):
     def _invoke(self, func, request, metadata=None):
         try:
             response = func(request, metadata=metadata)
-            error_code = response.status
-            error_message = ''
-        #Todo: AB#2325876: The error handling logic will be updated.
         except grpc.RpcError as rpc_error:
-            raise rpc_error
+            error_code = -1
+            error_message = rpc_error.details()
+            samps_per_chan_read = None
+            samps_per_chan_written = None
+            for entry in rpc_error.trailing_metadata() or []:
+                if entry.key == 'ni-error':
+                    value = entry.value if isinstance(entry.value, str) else entry.value.decode('utf-8')
+                    try:
+                        error_code = int(value)
+                    except ValueError:
+                        error_message += f'\nError status: {value}'
+                if entry.key == "samps_per_chan_read":
+                    samps_per_chan_read = entry.value
+                if entry.key == "samps_per_chan_written":
+                    samps_per_chan_written = entry.value
```

Comment:
1. gRPC metadata is like HTTP headers, so key names should use dashes, not underscores. 
2. We are prefixing NI-specific metadata with `ni-` to ensure that it doesn't conflict with metadata used by HTTP proxies and such.
3. These if-statements are mutually exclusive, so use `elif`.
4. The samps per chan read/written cases should decode `entry.value` if necessary and convert it to `int`.

Reply:
Note about decoding `bytes` -> `str`: only keys that end with `-bin` should contain `bytes`. However, mypy doesn't know that. `assert isinstance(entry.value, str)`  or `int(typing.cast(str, entry.value))` would also work.

Reply:
Updated the method as per the suggestion